coffee/lib/abstract-chosen.coffee

```diff
@@ -217,13 +217,15 @@ class AbstractChosen
       this.winnow_results_set_highlight()
 
   get_search_regex: (escaped_search_string) ->
-    regex_string = if @search_contains then escaped_search_string else "\\b#{escaped_search_string}\\w*\\b"
+    regex_string = if @search_contains then escaped_search_string else "(^|\\s)#{escaped_search_string}[^\\s]*"
```

Comment:
I believe word boundary matches more than just spaces, so we probably should extend this list.

Reply:
> Exactly which characters are word characters depends on the regex flavor you're working with. In most flavors, characters that are matched by the short-hand character class \w are the characters that are treated as word characters by word boundaries. Java is an exception. Java supports Unicode for \b but not for \w.
> http://regular-expressions.mobi/wordboundaries.html?wlr=1

Reply:
So probably `[^\w]` would work

Reply:
Oh wait, like you said, `\w` is ascii only, but still a list would be nice then.

Reply:
I think this can easily be demonstrated when adding a test for options with parentheses, “Cocos (Keeling) Islands” for example. Currently “keel” does give a match, but with this change it wouldn’t.

Reply:
@koenpunt good point!

First, @adunkman mentioned trying to use a list but switching back to a "good enough" approach when that didn't perform well.

With that i mind, I can think of a couple of approaches:

1. Stick with the "good enough" approach, but explicitly not try to cover all of the possible unicode line breaks. Maybe something like just combining the `\s` approach _with_ a `\b` approach (e.g. `(^|\\s|\\b)`). This should preserve all the ASCII word boundaries we expect, and catch whitespace boundaries which, while not perfect, still allows for better results for non-ASCII languages than what we have today

2. Actually try to adhere to all the unicode word boundary possibilities. Wikimedia has a [unicodejs](https://github.com/wikimedia/unicodejs) library that could be instructive. Specifically, its [`isbreak`](https://github.com/wikimedia/unicodejs/blob/master/src/unicodejs.wordbreak.js#L130-L258) method. This would increase code size by quite a bit, and probably decrease performance, since we wouldn't be in `re.exec`-land anymore.

I'd recommend (1), because (2) is a much bigger endeavor than (1) is. And (1) is still better than what we've got now.

Reply:
@koenpunt 

I paired with @adunkman on implementing idea (1) above and added [this commit](https://github.com/harvesthq/chosen/pull/2877/commits/fb6e071ec3992979b87ada5cdd318179bc116683). 

Let us know what you think!

Reply:
I think that approach is good enough, although I dont believe the tests around it are very thorough; only testing it when the special character is at the begin of the value. I think it would be good to have some more test cases where those characters appear in the middle and the end.